src/shared/components/query/QueryStore.ts

```diff
@@ -1684,3 +1658,48 @@ export const QueryStoreComponent = ComponentGetsStoreContext(QueryStore);
 
 const selectedGeneSets = '';
 export default selectedGeneSets;
+
+export function buildTableData(hierarchyData: any[]): Geneset[] {
+	let tableData: Geneset[] = [];

```

Comment:
i think this function makes an unnecessary copy of the data and iterates needlessly through all the items when what you're doing is just concatenating.  I think _.flatMap will actually accomplish this in one statement:
_.flatMap((node)=>return node.genesets||[]) might do the trick.

Reply:
ah, i see you are renaming gensesetId -> name.   seems like a big cost for this, no?